### PR TITLE
fix(use): create `package.json` when calling `corepack use` on empty dir

### PR DESCRIPTION
**Pull request by aduh95**
Fixes: https://github.com/nodejs/corepack/issues/347

### CODE REVIEW

File: sources/specUtils.ts [hunk 1]
```diff
@@ -96,10 +100,15 @@ export async function loadSpec(initialCwd: string): Promise<LoadSpecResult> {
...
+    try {
+      file = await fs.promises.open(manifestPath, `r`);
+    } catch (err) {
+      if ((err as NodeError)?.code === `ENOENT`) continue;
+      throw err;
+    }
 
-    const content = await fs.promises.readFile(manifestPath, `utf8`);
+    const content = await file.readFile(`utf8`);
+    await file.close();
```
Comment (arcanis) [inline sources/specUtils.ts:111]: Can't we just use `readFile` and assign `{}` to `data` wen catching `ENOENT`?

Comment (aduh95) [reply sources/specUtils.ts:111]: Sure, the only anti-pattern is to rely on `existsSync`, as its result might be outdated by the time the `readFile` calls is processed.

**Review by merceyz — COMMENTED** [review]
Could you add a test for what should happen when running something like
```sh
corepack use foo@1
mkdir src
cd src
corepack use foo@2
```

File: sources/commands/Use.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export class UseCommand extends BaseCommand {
     `,
     examples: [[
       `Configure the project to use the latest Yarn release`,
-      `corepack use 'yarn@*'`,
+      `corepack use 'yarn'`,
```
Comment (arcanis) [inline sources/commands/Use.ts:19]:
```suggestion
      `corepack use yarn`,
```

File: sources/specUtils.ts [hunk 3]
```diff
@@ -96,10 +99,12 @@ export async function loadSpec(initialCwd: string): Promise<LoadSpecResult> {
       continue;
 
     const manifestPath = path.join(currCwd, `package.json`);
-    if (!fs.existsSync(manifestPath))
-      continue;
-
-    const content = await fs.promises.readFile(manifestPath, `utf8`);
+    try {
+      content = await fs.promises.readFile(manifestPath, `utf8`);
+    } catch (err) {
```
Comment (arcanis) [inline sources/specUtils.ts:104]: nit: The `content` declaration should be right above the `try/catch`, so it's clearer where it's assigned
```suggestion
    let content: string;
    try {
      content = await fs.promises.readFile(manifestPath, `utf8`);
    } catch (err) {
```

**Review by arcanis — APPROVED** [review]
Couple of nits, but LGTM overall

File: sources/specUtils.ts [hunk 4]
```diff
@@ -88,6 +89,8 @@ export async function loadSpec(initialCwd: string): Promise<LoadSpecResult> {
     manifestPath: string;
   } | null = null;
 
+  let content: string;
+
```
Comment (aduh95) [inline sources/specUtils.ts:93]:
```suggestion
```

**Review by merceyz — APPROVED** [review]
LGTM.

NIT: This seems more like a new feature (`feat:`) than a bugfix (`fix:`).

**Comment by merceyz** [comment]
Right, in that case I agree, it's a bugfix 👍

**Pull request merged**